kerchunk/kerchunk_nos_roms.py

```diff
@@ -27,9 +56,51 @@ def generate_kerchunked_nc(region: str, bucket: str, key: str, dest_bucket: str,
     print(f'Successfully processed {url}')
 
 
-def parse_nos_sqs_message(sqs_payload: str):
+def generate_kerchunked_model_run(region: str, bucket: str, key: str):
+    '''
+    Generate or update the multizarr kerchunked aggregation for the model run that the specified file belongs to
     '''
-    Parse the SQS message from the NOS ROMS S3 bucket to extract the file metadata needed for kerchunking and uploading 
+    # For now SSL false is solving my cert issues **shrug**
+    fs_read = fsspec.filesystem('s3', anon=True, skip_instance_cache=True, use_ssl=False)
+    fs_write = fsspec.filesystem('s3', anon=False, skip_instance_cache=True, use_ssl=False)
+
+    try:
+        model_date, model_hour = parse_model_run_datestamp(key)
+        model_run_glob = generate_model_run_glob_expression(key, model_date, model_hour)
+    except Exception as e:
+        print(f'Failed to parse model run date and hour from key {key}: {e}. Skipping...')
+        return
+
+    model_run_files = fs_read.glob(f's3://{bucket}/{model_run_glob}')
+    model_run_files = sorted(['s3://'+f for f in model_run_files])
+
+    model_run_file_count = len(model_run_files)
+    print(f'Aggregating {model_run_file_count} model files for model run {model_date} t{model_hour}z...')
+
+    # TODO: Generalize this somehow? 
```

Comment:
I think it's fine for now. We might consider a config system later to help generalize things like filenames and dimension names for each dataset, which should be modifiable by an RA type person.

Reply:
Yeah I think so too. We could get slick with it, knowing now how argo can hook up so easily to cloud filesystems. But for a prototype it works well